src/url-utils.ts

```diff
@@ -30,6 +30,19 @@ const sharedRelativeUrlProperties =
 export type UrlString = string;
 
 /**
+ * Returns a URL with the basename removed from the pathname.  Strips the
+ * search off of the URL as well, since it will not apply.
+ */
+export function debaseUrl(href: UrlString): UrlString {
+  const u = url.parse(href);
+  if (typeof u.pathname === 'string') {
```

Comment:
Fair.  Weird issue-- typescript says it should be string | undefined, but the url.parse returns an object with nulls on it and doesn't let me assign nulls.  Anyways I corrected it to do a (u.pathname != null) which does the right thing and doesn't string match null, so it should be good enough now I think.